modules/au.com.trgtd.tr.task.messages/test/unit/src/au/com/trgtd/tr/task/messages/MessageParserTest.java

```diff
@@ -0,0 +1,158 @@
+package au.com.trgtd.tr.task.messages;
+
+import au.com.trgtd.tr.task.messages.MessageParser.Message;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Date;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.List;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+public class MessageParserTest {
+
+    // The message text stored in the sample messages.xml (pulled from the
+    // https://trgtd.com.au/messages.xml on 2023-01-14) is too long and
+    // cumbersome to validate verbatim.Simply verifying the length instead.
+    private static final int LENGTH_OF_SAMPLE_MESSAGE_TEXT = 431;
+
+    private final MessageParser parser = new MessageParser();
+
+    private final LocalDate dateMessage1 = LocalDate.of(2014, Month.MARCH, 27);
```

Comment:
dateMessage1, dateMessage2 - it's hard to read what the context of this dates.
This should be probably firstSampleDate and secondSampleDate.

Reply:
@RadekCap Not sure if I agree with your wording. Let me quickly explain my understanding:

Conceptually, we have an xml document with a set of messages. Each message has a date, a type, a text and an href.

Currently, in the MessageParserTest, we have two samples of such an xml document:

- sampleMessagesA: We use it to specify that depending on the date we use for parsing, we get both, none (not actually tested, but I might add that case too) or one of those messages.
- sampleMessagesB: We use it to differentiate between member users and non-member users only retrieving the messages intended for each group.

For simplicity, the first of the two messages of both samples are assigned one date (dateMessage1), the second of message of each sample has another date (dateMessage2). We could call it dateOfMessage2 if you prefer, but imho firstSampleDate has the wrong semantic.

Do you disagree?

Reply:
np, that's ok as is.

Reply:
Excellent, I'll merge then. Thanks for the thorough review. Very good improvements coming from it.